packages/rum-core/src/rawRumEvent.types.ts

```diff
@@ -209,3 +209,58 @@ export interface CommonContext {
   context: Context
   hasReplay?: true
 }
+
+export type RumEventDomainContext =
+  | RumViewEventDomainContext
+  | RumActionEventDomainContext
+  | RumFetchResourceEventDomainContext
+  | RumXhrResourceEventDomainContext
+  | RumOtherResourceEventDomainContext
+  | RumErrorEventDomainContext
+  | RumLongTaskEventDomainContext
+
+export interface RumViewEventDomainContext {
+  location: ReadonlyLocation
+}
+export interface RumActionEventDomainContext {
+  event?: Event
+}
+export interface RumFetchResourceEventDomainContext {
+  fetchInit?: RequestInit
+  fetchInput: RequestInfo
+  fetchResponse?: Response
+  fetchError?: Error
+  performanceEntry?: PerformanceEntryRepresentation
+}
```

Comment:
IMO, as it is not mixed with xhr anymore, fetch prefix could be omitted

Reply:
From a TS perspective, it is now clearer that we have a "fetch" context. But from a JS perspective, everything is still mixed together, so I think it's nice to have prefixes?

Reply:
could we avoid to have things mixed up?

Reply:
What do you have in mind? Something like this?

```ts
{
  longtask?: { performanceEntry: PerformanceEntryRepresentation },
  error?: { error: unknown },
  fetchResource?: { input: RequentInfo, init: RequestInit, response?: Response, error?: Error, performanceEntry?: PerformanceEntryRepresentation },
  xhrResource?: { xhr: XMLHttpRequest, performanceEntry?: PerformanceEntryRepresentation },
  otherResource?: { performanceEntry: PerformanceEntryRepresentation },
  autoAction?: { event: Event },
  view?: { location: Readonly<Location> }
}
```